templates/pre-install-secret.yaml

```diff
@@ -0,0 +1,22 @@
+# This shows a secret as a pre-install hook.
+# A pre-install hook is run before the rest of the chart is loaded.
+apiVersion: v1
+kind: Secret
+metadata:
+  name: pre-install-{{ template "fullname" . }}
+  labels:
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    app: {{ template "name" . }}
+  # This declares the resource to be a hook. By convention, we also name the
+  # file "pre-install-XXX.yaml", but Helm itself doesn't care about file names.
+  annotations:
+    "helm.sh/hook": pre-install
+type: Opaque
+data:
+  {{ if .Values.service.mysqlRootPassword }}
```

Comment:
I don't think `mysqlRootPassword` belongs here.

Reply:
Certainly not. Changed.